app/models/program_session.rb

```diff
@@ -46,6 +50,11 @@ def multiple_speakers?
     speakers.count > 1
   end
 
+  private
+
+  def destroy_speakers
+    speakers.each { |speaker| speaker.destroy unless speaker.proposals.exists? }
```

Comment:
This should only verify that the proposal_id exists on the speaker record and not that a related user through this speaker record has proposals.